### PR TITLE
cleanup tmp folder

### PR DESCRIPTION
**Pull request by aaaandrea**
Need to ensure we don't leave large bulky files around, so I added the use of [node-tmp](https://github.com/raszi/node-tmp) in order to cleanup post-tests. 
- [x] add tmp folder cleanup post bench
- [x] add tmp folder cleanup post tests

### CODE REVIEW

File: test/all.js [hunk 1]
```diff
@@ -3,8 +3,10 @@ const assert = require('assert');
 const tape = require('tape');
 const tmp = require('tmp');
 
+tmpOutput = tmp.tmpNameSync()
```
Comment (apendleton) [inline test/all.js]: * does this work (and get cleaned up afterwards)? Looking at the docs, I'd have thought you'd have had to actually create the thing with the library to get it to clean up after you (e.g., with `tmp.dir` or `tmp.dirSync`)
* can this have a `const` before it?
* nitpick: `output` as a name implies that it's our output. can it be, like, tmpDirectory or something?

File: test/all.js [hunk 2]
```diff
@@ -81,6 +83,7 @@ tape("FuzzyPhraseSetBuilder insertion and Set lookup", (t) => {
         "FuzzyPhraseSet fuzzy_match_prefix()"
     );
 
+    tmp.setGracefulCleanup();
```
Comment (apendleton) [inline test/all.js]: Not clear to me what the difference is in the docs between this and the remove callbacks that you get via `dir` or `dirSync` -- are you meant to have to call one of those as well? No need to change anything if not.

File: test/all.js (hunk 1 above)
Comment (aaaandrea) [reply test/all.js]: 👍 changed to `tmpDir`; updated to `const`

Also, I used `dirSync` which . generally works, but needed to use [rimraf](https://github.com/isaacs/rimraf) in order to rm the directory.

File: test/all.js (hunk 2 above)
Comment (aaaandrea) [reply test/all.js]: (addressed in other comment :) )

File: bench/fuzzysetbuilder.js [hunk 3]
```diff
@@ -78,5 +81,8 @@ rl.on('line', (line) => {
     console.log('     avg contains_prefix() lookup time: ' + (containsPrefixTotalTime/sampleSize) + 'ms');
     console.log('     avg fuzzy_match() lookup time: ' + (fuzzyMatchTotalTime/sampleSize) + 'ms');
     console.log('     avg fuzzy_match_prefix() lookup time: ' + (fuzzyMatchPrefixTotalTime/sampleSize) + 'ms');
+
+    rimraf(tmpDir.name);
```
Comment (apendleton) [inline bench/fuzzysetbuilder.js]: Bummer that this is necessary, but I guess `tmp` doesn't remove the directory's contents? I'm not sure why they would implement it that way...

**Pull request merged**